build-and-test.ps1

```diff
@@ -1,36 +1,56 @@
 #!/usr/bin/env pwsh
-[cmdletbinding()]
+[cmdletbinding(
+    DefaultParameterSetName = "BuildAndTest"
+)]
 param(
-    [ValidateSet('runtime', 'sdk', 'aspnet', 'wcf')]
+    [ValidateSet("runtime", "sdk", "aspnet", "wcf")]
     [string[]]$RepoFilter = @(),
     [string]$VersionFilter = "*",
     [string]$OSFilter = "*",
-    [string]$OptionalImageBuilderArgs,
-    [switch]$SkipTesting = $false
+    [Parameter(ParameterSetName = "Build")]
+    [switch]$BuildOnly,
+    [Parameter(ParameterSetName = "Test")]
+    [switch]$TestOnly,
+    [Parameter(ParameterSetName = "Build")]
+    [Parameter(ParameterSetName = "BuildAndTest")]
+    [string]$OptionalImageBuilderArgs
 )
 
+if ($PSCmdlet.ParameterSetName -eq "BuildAndTest") {
+    $build = $true
+    $test = $true
+}
+else {
+    $build = $BuildOnly
+    $test = $TestOnly
+}
+
 if ($RepoFilter.Count -eq 0) {
     $PathFilters = $null
-    $optionalTestArgs = ""
+    $testCategories = @("runtime", "sdk", "aspnet", "wcf")
```

Comment:
Couldn't this be left empty?  It would eliminate the maintenance of having the full list declared multiple times within the script.